packages/relay/src/lib/eth.ts

```diff
@@ -447,6 +447,41 @@ export class EthImpl implements Eth {
     }
   }
 
+  /**
+   * Gets a transaction by the provided hash
+   *
+   * @param hash
+   */
+  async getTransactionByHash(hash: string) {
+    this.logger.trace('getTransactionByHash(hash=%s)', hash);
+    const contractResult = await this.mirrorNodeClient.getContractResult(hash);
+    if (contractResult === null || contractResult.hash === undefined) {
+      return null;
+    }
+
+    return new Transaction({
+      accessList: contractResult.access_list,
+      blockHash: contractResult.block_hash.substring(0, 66),
+      blockNumber: contractResult.block_number,
+      chainId: contractResult.chain_id,
+      from: contractResult.from.substring(0, 42),
```

Comment:
Is the substring defensive or do we get longer strings from mirror node?  For the hashes we do get 384 bit hashes.

Reply:
This was my suggestion and just defensive. Actually checked MirrorNode response and min size is 40 and max is 42.
I can fix this in my following PR's